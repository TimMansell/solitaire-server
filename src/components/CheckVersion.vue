<template>
  <Toast
    v-if="!versionMatch"
    title="Update available"
    :msgs="[
      'A new version of the website is available.',
      'Refresh your browser or click on update',
    ]"
    btn-text="Update"
    :btn-click="refreshPage"
    data-test="version"
  />
</template>

<script>
import { mapGetters } from 'vuex';
import Toast from '@/components/Toast.vue';

export default {
  name: 'CheckVersion',
  components: {
    Toast,
  },
  computed: {
    ...mapGetters(['versionMatch']),
  },
  methods: {
    refreshPage() {
      window.location.href = '/';
    },
  },
};
</script>
