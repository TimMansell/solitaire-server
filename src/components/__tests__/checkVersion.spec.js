import { shallowMount } from '@vue/test-utils';
import CheckVersion from '@/components/CheckVersion.vue';

jest.mock('@/services/db');

describe('CheckVersion.vue', () => {
  it('matches version match snapshot', async () => {
    const wrapper = shallowMount(CheckVersion, {
      computed: {
        version: () => 'x.x.x',
      },
    });

    await wrapper.vm.$nextTick();

    expect(wrapper).toMatchSnapshot();
  });

  it('matches no version match snapshot', async () => {
    const wrapper = shallowMount(CheckVersion, {
      computed: {
        version: () => 'a.b.c',
      },
    });

    await wrapper.vm.$nextTick();

    expect(wrapper).toMatchSnapshot();
  });

  it('should match version and not show notification', async () => {
    const wrapper = shallowMount(CheckVersion, {
      computed: {
        version: () => 'x.x.x',
      },
    });

    await wrapper.vm.$nextTick();

    expect(wrapper.find('[data-test="version"]').exists()).toBe(false);
  });

  it('should not match version and show notification', async () => {
    const wrapper = shallowMount(CheckVersion, {
      computed: {
        version: () => 'a.b.c',
      },
    });

    await wrapper.vm.$nextTick();

    expect(wrapper.find('[data-test="version"]').exists()).toBe(true);
  });
});
