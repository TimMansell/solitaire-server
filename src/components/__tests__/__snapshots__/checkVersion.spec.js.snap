// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckVersion.vue matches no version match snapshot 1`] = `<toast-stub title="Update available" msgs="A new version of the website is available.,Refresh your browser or click on update" btntext="Update" btnclick="[Function]" data-test="version"></toast-stub>`;

exports[`CheckVersion.vue matches version match snapshot 1`] = ``;
